### PR TITLE
Add a CLI for rotating webhook secrets

### PR DESCRIPTION
**Pull request by jhrozek**
# Summary

Adds a minder-server subcommand that allows you to change minder webhook secrets
in github as a way of rotating the secret

Fixes: #2722

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

```
minder-server webhook update -p github
```

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [x] Added comments to complex or tricky code sections.
- [x] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

**Comment by JAORMX** [comment]
Will we need another k8s job to run this?

File: cmd/server/app/webhook_update.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	}
+	defer func(dbConn *sql.DB) {
+		err := dbConn.Close()
+		if err != nil {
+			zerolog.Ctx(ctx).Error().Err(err).Msg("error closing database connection")
+		}
+	}(dbConn)
+
+	store := db.NewStore(dbConn)
+	allProviders, err := store.GlobalListProvidersByName(ctx, providerName)
```
Comment (JAORMX) [inline cmd/server/app/webhook_update.go:87]: Should we instead iterate all providers, check that they implement github, and then attempt the migration? We're moving away from predetermined names and more into traits and classes

**Review by JAORMX — CHANGES_REQUESTED** [review]

File: docs/docs/run_minder_server/run_the_server.md [hunk 2]
```diff
@@ -249,6 +249,16 @@ webhook-config:
 
 After these steps, your Minder server should be ready to receive webhook events from GitHub, and add webhooks to repositories.
 
+In order to rotate webhook secrets, you can use the `minder-server` CLI tool to update the webhook secret.
+
+```bash
+minder-server webhook update -p github
+```
+
+Note that the command simply replaces the webhook secret in the configuration
```
Comment (eleftherias) [inline docs/docs/run_minder_server/run_the_server.md:258]: Isn't it the opposite? It seems like this command updates the secret in GitHub, but you will need to restart your Minder server to pick up the secret in the configuration.

**Review by eleftherias — COMMENTED** [review]
One comment on the documentation, but otherwise I tested it and it works great.

File: cmd/server/app/webhook_update.go (hunk 1 above)
Comment (jhrozek) [reply cmd/server/app/webhook_update.go:87]: We can do that, sure.

File: docs/docs/run_minder_server/run_the_server.md (hunk 2 above)
Comment (jhrozek) [reply docs/docs/run_minder_server/run_the_server.md:258]: gah, of course when I wrote configuration I meant github..

**Comment by jhrozek** [comment]
> Will we need another k8s job to run this?

yes, that's how I was thinking the rotation would work:
 - generate a new secret
 - set the new secret as the current one and the older one as a fallback
 - run this command as a job
 - monitor staging logs for the log message that says that a fallback secret was used to verify a webhook signature
 - if we see none, remove the fallback secret finally

File: cmd/server/app/webhook_update.go (hunk 1 above)
Comment (jhrozek) [reply cmd/server/app/webhook_update.go]: done!

**Review by JAORMX — APPROVED** [review]

**Pull request merged**